# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ShamirAli55/flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I picked **Content Refresh Prioritization**. The basic problem is that a website with thousands of pages can't review everything at once — you need some way to decide which pages to look at first. I wanted to see if the search performance data already in the dataset (impressions, position, CTR, content age) is enough to make that call more consistently than just eyeballing it.

It also felt like a concrete problem where a ranked output actually connects to something a real person would act on.

In [1]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("\nTrend Direction Distribution:")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:", round(df["ctr"].mean(), 4))
print("Average Content Age (days):", round(df["content_age_days"].mean(), 2))


FileNotFoundError: [Errno 2] No such file or directory: 'data/raw/content_refresh_anonymized.csv'

## 2. The question: decision, action, cost of a wrong call

**Decision:** Which pages should be flagged for a content review?

**Who acts on it:** A content manager. They get a ranked list and decide page by page whether to update. The model doesn't update anything — it just helps with triage.

**Cost of a wrong call:** Two kinds:
- Miss a declining page → it keeps sliding without anyone noticing. That's the more costly error because it's invisible.
- Flag a page that's fine → wastes a reviewer's time on something that doesn't need touching.

Neither error is free, but the first one compounds over time, so the ranking should lean toward catching actual decliners near the top.

In [ ]:
# Declining pages are over half the dataset — not a rare edge case
declining = (df["trend_direction"] == "down").sum()
total = len(df)
print(f"Declining pages: {declining} ({declining/total*100:.1f}%)")
print(f"Non-declining pages: {total - declining} ({(total-declining)/total*100:.1f}%)")


Declining pages: 16262 (54.2%)
Non-declining pages: 13738 (45.8%)


## 3. Quick look at the data (2-3 real numbers)

Three numbers that make this lane look worth the next several weeks:
- **16,262 pages** are already trending down — that's 54% of the dataset. There's clearly a real problem to prioritize.
- **Average content age: 256 days** — most of these pages haven't been touched in almost a year.
- **Average CTR: 0.51%** — low overall, which means there's headroom to improve if the right pages get refreshed.

In [ ]:
print("Total pages:", len(df))

print("\nTrend Direction Distribution")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:", round(df["ctr"].mean(), 4))

print("Average Content Age (days):", round(df["content_age_days"].mean(), 2))


Total pages: 30000

Trend Direction Distribution
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR: 0.5107
Average Content Age (days): 256.17


## 4. Careful words: what I can and can't claim

**Can claim:** associations between measured signals (impressions, CTR, position, content age) and the `trend_direction` label. Any finding is observed and directional — it's a pattern in this dataset, not a formula for rankings.

**Cannot claim:** why a page is declining, whether an update will fix it, or anything about Google's algorithm. The label itself (`trend_direction == 'down'`) is derived from the data, not an external ground truth.

The output is decision-support: it helps a human decide where to look, not a guarantee of what they'll find.

In [ ]:
# These columns encode the outcome — cannot be used as model features
leaky = [c for c in df.columns if "trend" in c]
print("Columns that must stay out of features (they reveal the label):")
print(leaky)


Columns that must stay out of features (they reveal the label):
['trend_direction', 'trend_pct']


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.